Sources/SemanticIndex/UpdateIndexStoreTaskDescription.swift

```diff
@@ -304,3 +346,56 @@ private func adjustSwiftCompilerArgumentsForIndexStoreUpdate(
   ]
   return result
 }
+
+/// Adjust compiler arguments that were created for building to compiler arguments that should be used for indexing.
+///
+/// This removes compiler arguments that produce output files and adds arguments to index the file.
+private func adjustClangCompilerArgumentsForIndexStoreUpdate(
+  _ compilerArguments: [String],
+  fileToIndex: DocumentURI
+) -> [String] {
+  let removeFlags: Set<String> = [
+    // Disable writing of a depfile
+    "-M",
+    "-MD",
+    "-MMD",
+    "-MG",
+    "-MM",
+    "-MV",
+    // Don't create phony targets
+    "-MP",
+    // Don't writ out compilation databases
+    "-MJ",
+    // Continue in the presence of errors during indexing
```

Comment:
Is this the reason? Or is it just that we don't have a decent timestamp to use for the "build session" here? Or does having this enable verification where we wouldn't otherwise?

Reply:
Honestly, I don’t know. I was trying to infer it but maybe I was wrong. I just removed the explanation for now.